### PR TITLE
[DataGrid] Add recipe showing how to toggle detail panels on row click

### LINKED ISSUE #7596: Show datagridpro details onRowClick

### PR DESCRIPTION
**Pull request by k-rajat19**
Fixes #7596 

[Preview](https://deploy-preview-14666--material-ui-x.netlify.app/x/react-data-grid/row-recipes/#toggling-detail-panels-on-row-click)

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/mui-x/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

### CODE REVIEW

**Comment by k-rajat19** [comment]
@romgrk should we add this recipe? ^_^

File: docs/next-env.d.ts
Comment (romgrk) [inline docs/next-env.d.ts:1]: What caused this change?

File: docs/data/data-grid/row-recipes/DetailPanelExpandOnRowClick.tsx [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+  const quantity = randomInt(1, 5);
+  return [...Array(quantity)].map((_, index) => ({
+    id: index,
+    name: randomCommodity(),
+    quantity: randomInt(1, 5),
+    unitPrice: randomPrice(1, 1000),
+  }));
+}
+
+const rows = [
```
Comment (romgrk) [inline docs/data/data-grid/row-recipes/DetailPanelExpandOnRowClick.tsx:114]: Any way to use one of the generated datasets? (employees or commodities)

Less lines of code is better.

Comment (k-rajat19) [reply docs/data/data-grid/row-recipes/DetailPanelExpandOnRowClick.tsx:114]: Initially, I tried to refactor all this but was unable to find a more concise way to do it
after seeing that same approach is used in other demos https://mui.com/x/react-data-grid/master-detail/
I ended up doing the same for this demo also.

File: docs/next-env.d.ts
Comment (k-rajat19) [reply docs/next-env.d.ts:1]: probably be caused while running some scripts, removed this change 👍

**Review by romgrk — APPROVED** [review]

**Pull request merged**